source/adiabatic_conditions/initial_profile.cc

```diff
@@ -72,7 +74,10 @@ namespace aspect
           MaterialModel::MaterialModelInputs<dim> in(1, n_compositional_fields);
           MaterialModel::MaterialModelOutputs<dim> out(1, n_compositional_fields);
           in.position[0] = representative_point;
-          in.temperature[0] = temperatures[i-1];
+          in.temperature[0] = (this->get_material_model().is_compressible())
```

Comment:
I would suggest to ask if the model uses adiabatic heating instead of if the material model is compressible. I know this would not be consistent with EBA (but for EBA we could always set up its own adiabatic conditions), but it would make a difference in incompressible models with adiabatic heating in our old (full) formulation. With your changes, these models would have a constant temperature in the adiabatic profile, although material would move on an adiabat when it is transported up and down, which means that the average temperature in the model would be very different from the adiabatic profile. In particular, if the density uses adiabatic_conditions, you would want your adiabatic conditions and the temperature in the model to be consistent. 
For all other cases (including BA, ALA, TALA), this would not make a difference, and for these models our adiabatic conditions should then also be consistent with what is assumed in the approximations.